llm/predictor.py

```diff
@@ -352,6 +353,50 @@ def _preprocess(self, source):
             inputs["attention_mask"] = self.attention_mask
             inputs["tgt_generation_mask"] = self.tgt_generation_mask
             inputs["tgt_pos"] = self.tgt_pos.numpy()
+        elif "bloom" in self.architectures:
+            inputs = dybatch_preprocess(self.tokenizer, source, self.config.max_length, self.architectures)
+            for i in range(inputs["input_ids"].shape[0]):
+                length = inputs["seq_len_encoder"][i][0]
+                self.attention_mask[i, 0, :length, :length] = paddle.tril(
+                    paddle.ones(shape=(length, length), dtype="float16")
+                )
+                self.arange_tensor_encoder[i, 0, :length] = paddle.arange(length).astype("float16")
+
+                self.tgt_generation_mask[i, 0, 0, :length] = paddle.ones(shape=[1, length], dtype="float16")
+            # alibi encoder
+            alibi_slopes = get_alibi_slopes(self.model_config.n_head)
+            inputs["position_ids"] = paddle.to_tensor(alibi_slopes, dtype="float32")
+
+            alibi = alibi_slopes[..., None] * self.arange_tensor_encoder
+            alibi = alibi[:, :, None, :]
+
+            if self.model_config.tensor_parallel_degree > 1:
+                block_size = self.model_config.n_head // self.model_config.tensor_parallel_degree
+                alibi = alibi[
+                    :,
+                    self.model_config.tensor_parallel_rank
+                    * block_size : (self.model_config.tensor_parallel_rank + 1)
+                    * block_size,
+                ]
+                alibi = alibi.reshape([inputs["input_ids"].shape[0], block_size, 1, self.config.max_length])
+                inputs["position_ids"] = inputs["position_ids"][
+                    self.model_config.tensor_parallel_rank
+                    * block_size : (self.model.config.tensor_parallel_rank + 1)
+                    * block_size
+                ]
+            alibi_encoder = alibi.expand(
+                [
+                    inputs["input_ids"].shape[0],
+                    self.model_config.n_head // self.model_config.tensor_parallel_degree,
+                    self.config.max_length,
+                    self.config.max_length,
+                ]
+            )
+            self.attention_mask = alibi_encoder + (1 - self.attention_mask) * -60000.0
```

Comment:
这里的-60000.0 应该需要使用 paddle.finfo() 来。
```suggestion
            self.attention_mask = alibi_encoder + (1 - self.attention_mask) * paddle.finfo(self.attention_mask.dtype).min
```